### ISSUE TITLE
Combobox dropdown always marks the first item as selected

### ISSUE THREAD

**Issue by rkovarik** [vaadin/framework#12557]
- Vaadin Framework version: 8.16

- Description of the bug:
1. Having a combobox with multiple pages in the dropdown.
2. Select an item from a next page from the dropdown (any page except the first one)
3. Open the dropdown again, you can see that the first item is marked as selected.
4. Confirm by enter. The value is changed without user realising.
 
- Expected behaviour: nothing is selected
- Actual behaviour: the first item is selected

- Minimal reproducible example:
```
ComboBox<Integer> comboBox = new ComboBox<>("", IntStream.range(0, 100).boxed().collect(Collectors.toList()));
comboBox.setEmptySelectionAllowed(false);
```

Workaround:
A] Educate users to not use ENTER to close the dropdown.
B] `comboBox.setScrollToSelectedItem(true);` Be aware:
> This requires finding the index of the item, which can be expensive in many large lazy loading containers.

**Comment by TatuLund** [comment]
Yes, this is a known behavior of Vaadin 8's ComboBox. I agree it is not the best possible. Thus when we rewrote ComboBox in our newer generation framework, we changed the behavior. Interesting thing was that some of our customers who have migrated from Vaadin 8 to Vaadin 14 or later, wanted to retain this behavior, so we have made add-on for our newer framework version, that has this Vaadin 8 behavior retained.

https://vaadin.com/directory/component/combo-box-with-vaadin-8-behavior

This raises the question, that if we would decide to "fix" this, or change this. Should we do it so that we introduce API that allows the select the behavior.
